onnxruntime/test/python/onnxruntime_test_python_autoep.py

```diff
@@ -0,0 +1,139 @@
+        input_name = sess.get_inputs()[0].name
+        res = sess.run([], {input_name: x})
+        output_expected = np.array([[1.0, 4.0], [9.0, 16.0], [25.0, 36.0]], dtype=np.float32)
+        np.testing.assert_allclose(output_expected, res[0], rtol=1e-05, atol=1e-08)
+
+    def test_example_plugin_ep_devices(self):
+        """
+        Test registration of an example EP plugin and retrieval of its OrtEpDevice.
+        """
+        ep_lib_path = "example_plugin_ep.dll"
+        ep_registration_name = "example_ep"
+
+        if sys.platform != "win32":
+            self.skipTest("Skipping test because it device discovery is only supported on Windows")
+
+        if not os.path.exists(ep_lib_path):
+            self.skipTest(f"Skipping test because EP library '{ep_lib_path}' cannot be found")
+
+        onnxrt.register_execution_provider_library(ep_registration_name, os.path.realpath(ep_lib_path))
+
+        ep_devices = onnxrt.get_ep_devices()
+        has_cpu_ep = False
+        test_ep_device = None
+        for ep_device in ep_devices:
+            ep_name = ep_device.ep_name
+
+            if ep_name == "CPUExecutionProvider":
+                has_cpu_ep = True
+            if ep_name == ep_registration_name:
+                test_ep_device = ep_device
+
+        self.assertTrue(has_cpu_ep)
+        self.assertIsNotNone(test_ep_device)
+
+        # Test the OrtEpDevice getters. Expected values are from /onnxruntime/test/autoep/library/example_plugin_ep.cc
+        self.assertEqual(test_ep_device.ep_vendor, "Contoso")
+
+        ep_metadata = test_ep_device.ep_metadata
+        self.assertEqual(ep_metadata["version"], "0.1")
+
+        ep_options = test_ep_device.ep_options
+        self.assertEqual(ep_options["run_really_fast"], "true")
+
+        # The CPU hw device info will vary by machine so check for the common values.
+        hw_device = test_ep_device.device
+        self.assertEqual(hw_device.type, onnxrt.OrtHardwareDeviceType.CPU)
+        self.assertGreaterEqual(hw_device.vendor_id, 0)
+        self.assertGreaterEqual(hw_device.device_id, 0)
+        self.assertGreater(len(hw_device.vendor), 0)
+
+        hw_metadata = hw_device.metadata
+        self.assertGreater(len(hw_metadata), 0)  # Should have at least SPDRP_HARDWAREID on Windows
+
+        # Test adding this EP plugin's OrtEpDevice to the SessionOptions.
+        sess_options = onnxrt.SessionOptions()
+        with self.assertRaises(InvalidArgument) as context:
+            # Will raise InvalidArgument because ORT currently only supports provider bridge APIs.
+            # Actual plugin EPs will be supported in the future.
+            sess_options.add_provider_for_devices([test_ep_device], {"opt1": "val1"})
+        self.assertIn("EP is not currently supported", str(context.exception))
+
+        onnxrt.unregister_execution_provider_library(ep_registration_name)
+
+
+if __name__ == "__main__":
+    unittest.main(verbosity=1)
```

Comment:
was this for debugging or do you want to leave it at verbosity=1

Reply:
removed